### PR TITLE
feat(BUILD-5293): add support for version with prefix

### PR DESCRIPTION
**Pull request by matemoln**

### CODE REVIEW

**Review by jayadeep-km-sonarsource — APPROVED** [review]

File: src/utils/version_helper.py [hunk 1]
```diff
@@ -2,44 +2,50 @@
 
 
 class VersionHelper:
-    VERSION_NUMBER_PLUS_SIGN_REGEX = r"^(\d+)\.(\d+)\.(\d+)\+(\d+)$"
+    VERSION_REGEX = (
+        r'^'                  # Start of the string
```
Comment (sebastienvermeille) [inline src/utils/version_helper.py:6]: Comments should be avoided as much as possible as they slow down the read of the code. 

If the code itself is not explicit enough and "can't" be that explicit then sure use comments (https://blog.cleancoder.com/uncle-bob/2017/02/23/NecessaryComments.html)

These comments are basically teaching "REGEX" -> Each DEV has to know about REGEX  and there are a lot of Books and documentation on the internet about it.

A REGEX starts with `^` and end with a `$` this is part of REGEX definition. (As an example, to me it looks like commenting a `;` in the end of a java line, the comment is not wrong but meh... what is the value of this comment, if I program in JAVA I am supposed to know that each lines end with a `;`.

A bit same here: "non-capturing group" "captured group" this is REGEX notions. I would not add it.

Talking about accepted parts of the regex well why not.. but the regex defines it already.


TLDR; I would not comment this regex at all, I would document the wanted behaviour in the function using :
1) Either in the doc of the function (comment doc) with a set of examples of accepted values + the general format evtl
2) And/Or in the tests, remember, tests are the best doc, they are always up to date! (**not** comments)
(this is what you did: tests + doc -> enough IMHO)

(Up to you to change it or not)

File: src/utils/version_helper.py [hunk 2]
```diff
@@ -2,44 +2,50 @@
...
-
-        if not re.match(VersionHelper.VERSION_NUMBER_PLUS_SIGN_REGEX, version):
-            return False
-
-        return VersionHelper.is_valid_sonar_version(VersionHelper._sanitize_version(version))
-
-    @staticmethod
-    def is_valid_version(version: str) -> bool:
-        return VersionHelper.is_valid_sonar_version(version) or VersionHelper.is_valid_plus_signed_version(version)
+    def validate_version(version: str) -> None:
```
Comment (sebastienvermeille) [inline src/utils/version_helper.py:15]: You decided to remove `is_valid_plus_signed_version`, that's your right, however here we are loosing fundamental comments: WHY do we accept `+` -> 

```
     # This is an explicit requirement for project SLVSCODE
        # see https://sonarsource.atlassian.net/browse/BUILD-4915 for more details       
```

Such comments are literally gold, because when you refactor code, in order to simplify the program (part of clean code), you need these information.

Without such comments, you just check 3 projects (i.e: sonar-dummy sonar-dummy-oss well they don't use any `+` let's remove it it's making things complicated for nothing`) -> boom you accidentally broke a feature.

Or at very least a specific test should be there with these comments to say why we accept + and now that you add other variants, please add a ticket reference about WHY we support them.

This is very important to avoid future regression.

**Review by sebastienvermeille — DISMISSED** [review]
Well done! I just some shared some notes about deleted comments and comments missing (and eventually some to remove based on your appreciation)

File: src/utils/version_helper.py (hunk 1 above)
Comment (jayadeep-km-sonarsource) [reply src/utils/version_helper.py:6]: > Comments should be avoided as much as possible as they slow down the reading of the code.

I can't entirely agree, especially in this context. If I need to spend more than 10 seconds to understand one line of code, then having a comment is valuable. Regex is the perfect example of this. For example, I know that `d` means a digit, but understanding `r"^(\d+)\.(\d+)\.(\d+)\+(\d+)$"`  requires parsing all the brackets and symbols in my head. I would prefer having a one liner comment with an example in this case.

Comment (matemoln) [reply src/utils/version_helper.py]: I wanted to be verbose because of the unintuitive nature of regex, but I might have gone too far.
Thanks for the feedback! I cleaned up the comments a little bit.

File: src/utils/version_helper.py (hunk 2 above)
Comment (matemoln) [reply src/utils/version_helper.py:14]: Added the links back and done the same at https://github.com/SonarSource/gh-action_release/pull/217

**Review by jayadeep-km-sonarsource — APPROVED** [review]
